### ISSUE TITLE
Component skeletons (loading states) need a11y improvements for screen reader users

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#4310]
ref #4176 
ref #4307 
attn @snidersd 

I'm specifically referring to our React package.

The loading states for our components (referred to as skeletons) are not accessible for screen reader users. I'm referencing requirements around status messages [found here](https://www.ibm.com/able/guidelines/ci162/status_messages_71.html).

Improvements:
- add `tabindex="0"`
- add `role="status"`
- add `aria-live="assertive"`
- add `aria-label="loading {components name}"`

Components with Skeletons:
- [ ] Accordion
- [ ] Breadcrumb
- [ ] Button
- [x] Checkbox
- [ ] CodeSnippet
- [ ] DataTable
- [ ] DatePicker
- [ ] Dropdown
- [ ] FileUploader
- [ ] Icon
- [x] NumberInput
- [ ] ProgressIndicator
- [ ] RadioButton
- [ ] Search
- [x] Select
- [ ] Text
- [x] Slider
- [ ] StructuredList
- [ ] Tabs
- [ ] Tag
- [x] TextArea
- [x] TextInput
- [ ] Toggle
- [ ] ToggleSmall

**Comment by jendowns** [comment]
In #4307, I've addressed these attributes for the following skeleton components:
* Checkbox
* NumberInput
* Select
* Slider
* TextArea
* TextInput

**Comment by dakahn** [comment]
Closing this issue. We're exploring alternate solutions to this problem rather than addressing loading component by component 👍
